src/plugins/vis_augmenter/public/saved_augment_vis/saved_augment_vis.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { get, isEmpty } from 'lodash';
+import {
+  SavedObjectLoader,
+  SavedObjectOpenSearchDashboardsServices,
+} from '../../../saved_objects/public';
+import { createSavedAugmentVisClass } from './_saved_augment_vis';
+import { VisLayerTypes } from '../../common';
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface SavedObjectOpenSearchDashboardsServicesWithAugmentVis
+  extends SavedObjectOpenSearchDashboardsServices {}
+export type SavedAugmentVisLoader = ReturnType<typeof createSavedAugmentVisLoader>;
+export function createSavedAugmentVisLoader(
+  services: SavedObjectOpenSearchDashboardsServicesWithAugmentVis
+) {
+  const { savedObjectsClient } = services;
+
+  class SavedObjectLoaderAugmentVis extends SavedObjectLoader {
+    mapHitSource = (source: Record<string, any>, id: string) => {
+      source.id = id;
+      source.visId = get(source, 'visReference.id', '');
+
+      if (isEmpty(source.visReference)) {
+        source.error = 'visReference is missing in augment-vis saved object';
+      }
+      if (isEmpty(source.visLayerExpressionFn)) {
+        source.error = 'visLayerExpressionFn is missing in augment-vis saved object';
+      }
+      if (!(get(source, 'visLayerExpressionFn.type', '') in VisLayerTypes)) {
+        source.error = 'Unknown VisLayer expression function type';
+      }
+
+      delete source.visReference;
+      delete source.visName;
```

Comment:
why remove these attributes before return?

Reply:
I was thinking these would become unnecessary fields once the vis ID has been extracted, similar to what's done in [injectReferences()](https://github.com/ohltyler/OpenSearch-Dashboards/blob/add-saved-obj/src/plugins/vis_augmenter/public/saved_augment_vis/saved_augment_vis_references.ts#L53).

But given the fn signature is only about adding fields to the source obj, and for consistency with other implementations of `mapHitSource()`, I agree that these shouldn't be removed. I'll update

Reply:
Fixed in [this commit](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3109/commits/f1b08adfe2aebed6cab9755980e5f909877f0aa4)